libs/application-generic/src/services/resource-validator.service.ts

```diff
@@ -0,0 +1,93 @@
+import { Injectable } from '@nestjs/common';
+import { NotificationTemplateRepository } from '@novu/dal';
+import { FeatureFlagsKeysEnum } from '@novu/shared';
+
+import { GetFeatureFlag, NotificationStep } from '../usecases';
+import { ApiException } from '../utils/exceptions';
+
+@Injectable()
+export class ResourceValidatorService {
+  private readonly MAX_STEPS_PER_WORKFLOW = 10;
+  private readonly OLD_WORKFLOWS_LIMIT = 1000;
+  private readonly NEW_WORKFLOWS_LIMIT = 100;
+
+  constructor(
+    private notificationTemplateRepository: NotificationTemplateRepository,
+    private getFeatureFlag: GetFeatureFlag,
+  ) {}
+
+  async validateStepsCount(environmentId: string, steps: NotificationStep[]) {
+    const isWorkflowLimitEnabled = await this.getFeatureFlag.execute({
+      key: FeatureFlagsKeysEnum.IS_WORKFLOW_LIMIT_ENABLED,
+      environmentId,
+      organizationId: 'system',
+      userId: 'system',
+    });
+
+    if (!isWorkflowLimitEnabled) {
+      return;
+    }
+
+    if (steps.length > this.MAX_STEPS_PER_WORKFLOW) {
+      throw new ApiException({
+        message: `Workflow steps limit exceeded. Maximum allowed steps is ${this.MAX_STEPS_PER_WORKFLOW}, but got ${steps.length} steps.`,
+        providedStepsCount: steps.length,
+        maxSteps: this.MAX_STEPS_PER_WORKFLOW,
+      });
+    }
+  }
+
+  async validateWorkflowLimit(environmentId: string) {
+    const isWorkflowLimitEnabled = await this.getFeatureFlag.execute({
+      key: FeatureFlagsKeysEnum.IS_WORKFLOW_LIMIT_ENABLED,
+      environmentId,
+      organizationId: 'system',
+      userId: 'system',
+    });
+
+    if (!isWorkflowLimitEnabled) {
+      return;
+    }
+
+    const enforcementDate = await this.getFeatureFlag.date({
+      key: FeatureFlagsKeysEnum.DATE_WORKFLOW_LIMIT_ENFORCEMENT,
+      environmentId,
+      organizationId: 'system',
+      userId: 'system',
+    });
```

Comment:
self note: to remove this logic.
the date feature flag is unnecessary, we can keep is enabled, and a static date will be enough

Reply:
@djabarovgeorge can we make this work with a single feature flag (IS_WORKFLOW_MAX_LIMIT_ENABLED) and add date conditions in Launch Darkly by passing the organization created_at context to LD?

Reply:
interesting 🙃

Reply:
Added the support for one boolean FF to rule them all as mentioned in the DM, you can see the example of the needed configuration there as well.